Add retry options to nightly pipeline

sf1919 opened the pull request:
**Description of work.**
Added retry options to two stages of the nightly pipeline in an effort to reduce failed builds. Specifically:-
- package conda
- Package: Workbench


**To test:**
Check the build from this pipeline - https://builds.mantidproject.org/job/build_packages_from_branch/47/
Check that all tests pass
Code review

*There is no associated issue.*

*This does not require release notes* because **this is an update to the CI pipeline**


<!-- Ensure the base of this PR is correct (e.g. release-next or main)
Finally, don't forget to add the appropriate labels, milestones, etc.!  -->

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

sf1919 committed:
Added retry option

Added the retry option to stages
- package conda
- Package: Workbench

martyngigg commented on buildconfig/Jenkins/Conda/nightly_build_and_deploy.jenkinsfile:
```diff
@@ -111,7 +111,10 @@ pipeline {
             }
             agent { label "${PLATFORM}" }
-            options { timestamps () }
+            options {
```
This looks good and matches what the documentation suggests is the right way to go.

Can I just be picky about the indentation from a readability point of view then can we indent the lines inside the block and leave the closing brace under the `o` as:

```groovy
options {
  timestamps()
  retry(3)
}
```

sf1919 committed:
Changes after review

- improved formatting for easier reading
- added a retry option to publishing stage

martyngigg reviewed (approved):
This looks great. Good to go 👍🏻
